lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -921,6 +1090,36 @@ func (c *ContainerProxy) createNewAttachClientWithTimeouts(connectTimeout, respo
 	return plClient, transport
 }
 
+func (c *ContainerProxy) createGzipTarClient(connectTimeout, responseTimeout, responseHeaderTimeout time.Duration) (*client.PortLayer, *httpclient.Transport) {
+
+	r := rc.New(c.portlayerAddr, "/", []string{"http"})
+	transport := &httpclient.Transport{
+		ConnectTimeout:        connectTimeout,
+		ResponseHeaderTimeout: responseHeaderTimeout,
+		RequestTimeout:        responseTimeout,
+	}
+
+	r.Transport = transport
+
+	plClient := client.New(r, nil)
+	r.Consumers["application/octet-stream"] = runtime.ByteStreamConsumer()
+	r.Producers["application/octet-stream"] = runtime.ByteStreamProducer()
+
+	r.Consumers["application/x-tgz"] = runtime.ConsumerFunc(func(rdr io.Reader, data interface{}) error {
+		bsc := runtime.ByteStreamConsumer()
```

Comment:
I looked at the code for ByteStreamConsumer(), it's just returning an interface with a go function.  This should be fine.  I also don't like global vars if I can help it.